### ISSUE TITLE
gstoraster: Ghostscript runs to long while converting djvu files to printer`s format

### ISSUE THREAD

**Issue by valia0906** [OpenPrinting/cups-filters#354]
https://bugs.launchpad.net/ubuntu/+source/cups-filters/+bug/1920730

**Comment by tillkamppeter** [comment]
@valia0906, could you attach a sample djvu file and also instructions how you print the djvu file, so that we can reproduce the problem and perhaps improve the patch somewhat? This is especially important as we want to carry on the CUPS filters when CUPS will not use PPD files any more.

If you are the poster of the Ubuntu bug or if you have at least tried out the patch posted there and found out that it solves the problem. could you try the following:

As the Ghostscript call the patch is modifying does not more than counting the pages (there is another one for the actual rasterization) could you also try to modify the patch by not calling cupsRasterInterpretPPD() for extracting the default page size from the PPD and instead, simply insert arbitrary height and width, like "-dDEVICEWIDTHPOINTS=612 -dDEVICEHEIGHTPOINTS=792"? This would allow you to remove the "if (ppd)" and so your fix would also work for using the filter without PPD.

Could you try this out? Do you still get fast execution and correct page counts (= filter outputs all pages)?

@surajkulriya, if you are able to reproduce the original problem (having an input file which makes the original filter taking to long but the patch solving this) and did not simply grab the patch, could you also try out what I am suggesting above? This would also be a first step for your GSoC project.

**Comment by tillkamppeter** [comment]
@surajkulriya I got a test file (see Ubuntu bug report) and tried it as I described there. I found that the best solution is to simply add `-dDEVICEWIDTHPOINTS=1 -dDEVICEHEIGHTPOINTS=1` to the command line:
```
gs -q -dNOPAUSE -dBATCH -sDEVICE=bbox -dDEVICEWIDTHPOINTS=1 -dDEVICEHEIGHTPOINTS=1 sample2.pdf 2>&1 | grep -c HiResBoundingBox
```
So no `if (ppd)` and no call of `cupsRasterInterpretPPD()` is needed, and the fix also applies to PPD-less use of the filter.

**Comment by surajkulriya** [comment]
@tillkamppeter  sir, 
Thanks for the update, I have made changes in the latest commit.

**Comment by tillkamppeter** [comment]
Fixed by pull request #371.

Thank you very much.
